src/ucs/type/spinlock.h

```diff
@@ -40,50 +51,91 @@ typedef struct ucs_recursive_spinlock {
 } ucs_recursive_spinlock_t;
 
 
-static ucs_status_t ucs_spinlock_init(ucs_spinlock_t *lock, int flags)
+void ucs_recursive_spinlock_destroy(ucs_recursive_spinlock_t *lock);
+
+/**
+ * Spinlock implementation section
+ */
+static UCS_F_ALWAYS_INLINE void ucs_spinlock_init(ucs_spinlock_t *lock)
 {
-    int ret, lock_flags;
+    lock->lock = UCS_SPINLOCK_FREE;
+}
 
-    if (flags & UCS_SPINLOCK_FLAG_SHARED) {
-        lock_flags = PTHREAD_PROCESS_SHARED;
-    } else {
-        lock_flags = PTHREAD_PROCESS_PRIVATE;
+static UCS_F_ALWAYS_INLINE void ucs_spinlock_destroy(ucs_spinlock_t *lock)
+{
+}
+
+static UCS_F_ALWAYS_INLINE int ucs_spin_try_lock(ucs_spinlock_t *lock)
+{
+    int res;
+
+    res = ucs_atomic_cswap32(&lock->lock, UCS_SPINLOCK_FREE,
+                             UCS_SPINLOCK_BUSY) == UCS_SPINLOCK_FREE;
+
+#if defined(__powerpc64__)
+    /* barrier on locked spin */
```

Comment:
comment not clear - why is it needed? for memory ordering? ucs_atomic_cswap32 is not enough?